services/management/grpc/actions_server.go

```diff
@@ -37,8 +38,21 @@ type actionsServer struct {
 	l  *logrus.Entry
 }
 
+// structure to keep id with timestamp
+type idWithTimeStamp struct {
+	id        string
+	timestamp int64
+}
+
 var pmmAgent2100 = version.MustParse("2.10.0-HEAD") // TODO: Remove HEAD later once 2.11.0 is released.
 
+// ****** TBD: Define where the period will be stored or configured
+// Period that pt_summary action in seconds is cached for. After this time the pt-summary will be refreshed again.
+const ptSummaryRefreshPeriod = 60
+
+// Dictionary that of agent_id againts a structure of action_id and timestamp
+var dicPtSummaryLastAction = make(map[string]idWithTimeStamp)
```

Comment:
can we include cleanup logic?

Reply:
and maybe it's better to make it field of actionsServer structure?

Reply:
And it's not thread-safe

Reply:
To save time I suggest we have a meeting. Can you set up a meeting where the right people will be?